extensions/ql-vscode/src/queryserver-client.ts

```diff
@@ -1,5 +1,7 @@
 import * as cp from 'child_process';
-import { DisposableObject } from 'semmle-vscode-utils';
+// Import from the specific module within `semmle-vscode-utils`, rather than via `index.ts`, because
+// we avoid taking an accidental runtime dependency on `vscode` this way.
+import { DisposableObject } from 'semmle-vscode-utils/out/disposable-object';
```

Comment:
I'm a little confused by this. Depending on `/out/` seems abstraction-boundary-breaking in a way that makes me feel worried. If we really wanted to cut this dependency, could we instead just refactor `disposable-object` to refer to a pared down interface we define ourselves like
```
interface SimplyDisposable {
  dispose(): void;
}
```
?
Separately, on the very next line we depend on `{ Disposable } from 'vscode'`, so what's being saved?

Reply:
There is a unit test that tries to run code from `queryserver-cli.ts` outside of VSCode. If we just do `import { DisposableObject } from 'semmle-vscode-utils'`, that imports the `index.ts` of `semmle-vscode-utils`, which in turn imports not just the definition of `DisposableObject`, but also other code. `DisposableObject` only depends on `Disposable`, which is merely a TypeScript type that gets erased when transpiled to JavaScript. The other code in `semmle-vscode-utils`, however, has actual implementation dependencies on executable code exported from `vscode`, and therefore forces a runtime dependency on `vscode`. By importing only that single file from within the module, we avoid importing the other code from `semmle-vscode-utils`, and thus avoid the runtime dependency.

A cleaner solution would probably be to move `DisposableObject` into a package that has only a dev dependency on `vscode`, or simply defines its own `Disposable`. I'm happy to do this, either in this PR or in a follow up.

Reply:
Here's the failure this causes: https://github.com/dbartol/vscode-codeql/runs/490822526

Reply:
Ah, sounds good. didn't realize it was actually *causing* test failures, rather than just being a cleanliness issue. Filed https://github.com/github/vscode-codeql/issues/268.